python/llm/example/GPU/QLoRA-FineTuning/qlora_finetuning.py

```diff
@@ -61,6 +61,17 @@
         task_type="CAUSAL_LM"
     )
     model = get_peft_model(model, config)
+
+    for name, module in model.named_modules():
+        if isinstance(module, LoraLayer):
+            module = module.to(torch.bfloat16)
+        if 'norm' in name:
+            module = module.to(torch.float32)
+        if 'lm_head' in name or 'embed_tokens' in name:
+            if hasattr(module, 'weight'):
+                if module.weight.dtype == torch.float32:
+                    module = module.to(torch.bfloat16)
```

Comment:
put these into a utility function

Reply:
Have put this logic into `cast_lora_weight` function.